test/recipes/15-test_rsaoaep.t

```diff
@@ -0,0 +1,156 @@
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use strict;
+use warnings;
+
+use OpenSSL::Test qw(:DEFAULT data_file bldtop_dir srctop_file srctop_dir bldtop_file);
+use OpenSSL::Test::Utils;
+use File::Compare qw/compare_text/;
+
+BEGIN {
+setup("test_rsaoaep");
+}
+use lib srctop_dir('Configurations');
+use lib bldtop_dir('.');
+use platform;
+
+my $no_fips = disabled('fips') || ($ENV{NO_FIPS} // 0);
+
+plan tests =>
+    ($no_fips ? 0 : 1)          # FIPS install test
+    + 9;
+
+my @prov = qw( -provider_path providers -provider null -provider_config none -provider default );
+my $msg_file = data_file("plain_text");
+my $enc_file = "enc.bin";
+my $enc2_file = "enc2.bin";
+my $dec_file = "dec.txt";
+my $key_file = srctop_file("test", "testrsa.pem");
+
+unless ($no_fips) {
+    my @prov = qw( -provider_path providers -provider null
+                  -provider_config ./test/fips.cnf -provider fips );
+    my $infile = bldtop_file('providers', platform->dso('fips'));
+
+    ok(run(app(['openssl', 'fipsinstall',
+                '-out', bldtop_file('providers', 'fipsmodule.cnf'),
+                '-module', $infile,
+                '-provider_name', 'fips', '-mac_name', 'HMAC',
+                '-macopt', 'digest:SHA256', '-macopt', 'hexkey:00',
+                '-section_name', 'fips_sect'])),
+       "fipsinstall");
+}
+
+ok(run(app(['openssl', 'pkeyutl',
+            @prov,
+            '-encrypt',
+            '-in', $msg_file,
+            '-inkey', $key_file,
+            '-pkeyopt', 'pad-mode:oaep',
+            '-pkeyopt', 'oaep-label:123',
+            '-pkeyopt', 'digest:sha1',
+            '-pkeyopt', 'mgf1-digest:sha1',
+            '-out', $enc_file])),
+   "RSA OAEP Encryption");
+
+ok(run(app(['openssl', 'pkeyutl',
+            @prov,
+            '-decrypt',
+            '-inkey', $key_file,
+            '-pkeyopt', 'pad-mode:oaep',
+            '-pkeyopt', 'oaep-label:123',
+            '-pkeyopt', 'digest:sha1',
+            '-pkeyopt', 'mgf1-digest:sha1',
+            '-in', $enc_file,
+            '-out', $dec_file]))
+    && compare_text($dec_file, $msg_file) == 0,
+    "RSA OAEP Decryption");
+
+ok(run(app(['openssl', 'pkeyutl',
+            @prov,
+            '-encrypt',
+            '-in', $msg_file,
+            '-inkey', $key_file,
+            '-pkeyopt', 'pad-mode:oaep',
+            '-pkeyopt', 'oaep-label:123',
+            '-pkeyopt', 'digest:sha1',
+            '-pkeyopt', 'mgf1-digest:sha1',
+            '-out', $enc2_file]))
+    && compare_text($enc2_file, $enc_file) != 0,
+   "RSA OAEP Encryption should generate different encrypted data");
+
+ok(run(app(['openssl', 'pkeyutl',
+            @prov,
+            '-decrypt',
+            '-inkey', $key_file,
+            '-pkeyopt', 'pad-mode:oaep',
+            '-pkeyopt', 'oaep-label:123',
+            '-in', $enc_file,
+            '-out', $dec_file]))
```

Comment:
Please use different output files for every run.  That allows looking at them after the test is run too re-play or find out what went wrong when a test fails (the output files of this test are found in `test-runs/test_rssaoaep`)

Reply:
Done - shuffled the tests around also so they are all grouped together for the file that is used.